code/game/machinery/bots/secbot.dm

```diff
@@ -384,6 +384,22 @@ Auto Patrol: []"},
 		target = user
 		mode = BOT_HUNT
 
+/obj/machinery/bot/secbot/Crossed(atom/movable/AM)
+	if(ismob(AM) && target)
+		var/mob/living/carbon/C = AM
+		if(!istype(C) || !C || in_range(src, target))

```

Comment:
Unnecessary sanity checks make for ugly code

if ever crossed was called when you're not on top of beepsky that would be a bug that needs fixing. This check would hide that situation.